content/articles/youtube-video-downloader-using-python/index.md

```diff
@@ -0,0 +1,139 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /youtube-video-downloader-using-python/
+title: Creating a YouTube Video Downloader using Python
+description: In this article we will build a YouTube downloader using Python.
+author: victor-kamau
+date: 2022-03-10T00:00:00-18:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+ - url: /engineering-education/youtube-video-downloader-using-python/hero.png
+   alt: Python Video Downloader example image
+---
+
+YouTube is an internationally known website for sharing videos. Unfortunately, when you use an online downloader to download a video or try to get it from a random website, you run the danger of your personal information being stolen. 
+
+<!-- more -->
+
+### Introduction
+The task of downloading videos is incredibly straightforward, efficient, and safe while using the Python Tkinter module. Moreover, it gets the video directly from YouTube for you.
+
+### Table of Contents
+- [Introduction](#introduction)
+- [Table of Contents](#table-of-contents)
+- [Requirements](#requirements)
+- [Installations](#installations)
+  - [Installing PyTube module](#installing-pytube-module)
+  - [Install **Tkinter**](#install-tkinter)
+- [Building the API/GUI](#building-the-apigui)
+  - [1.Import Libraries needed in the project](#1import-libraries-needed-in-the-project)
+  - [2. Create the API window](#2-create-the-api-window)
+  - [3. Create the link entry](#3-create-the-link-entry)
+  - [4. Create the download button](#4-create-the-download-button)
+- [Creating the download function](#creating-the-download-function)
+- [Conclusion](#conclusion)
+
+### Requirements
+To develop a video downloader, you should have;
+- Prior and a better understanding of Python language
+- Python package installed on your PC
+- An editor of choice
+- Internet connection for API testing
+- **Tkinter** is a Python module for creating GUIs.
+- **Pytube** downloads the video from YouTube.
+
+### Installations
+Requirements need to be ready before you start your development process. A Python environment is a foremost requirement for this project. You can follow [these directives](https://www.section.io/engineering-education/a-beginners-guide-to-python/) to make set-up a Python environment on your PC. A crucial module needs to be installed, which is the beginning of the development process.

```

Comment:
But [this tutorial](https://www.section.io/engineering-education/a-beginners-guide-to-python/) does not show the readers how to set up a Python env.